src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -0,0 +1,167 @@
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveWheelSpeeds;
+import frc.robot.components.Pigeon;
+import frc.robot.subsystems.MoveableSubsystem;
+import static frc.robot.Robot.robotConstants;
+
+/**
+ * Creates a new DrivetrainSubsytem.
+ */
+public class Drivetrain implements MoveableSubsystem {
+
+  private DifferentialDriveKinematics kinematics;
+  private final DifferentialDriveOdometry odometry;
+
+  private WPI_TalonFX leftRearTalon;
+  private WPI_TalonFX leftMiddleTalon;
+  private WPI_TalonFX leftFrontTalon;
+  private WPI_TalonFX rightRearTalon;
+  private WPI_TalonFX rightMiddleTalon;
+  private WPI_TalonFX rightFrontTalon;
+  private WPI_TalonSRX rightEncoder;
+  private WPI_TalonSRX leftEncoder;
+  private Pigeon gyro;
+
+  private DifferentialDrive drivetrain;
+
+  public Drivetrain() {
+    leftRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    leftMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_MIDDLE_TALON);
+    leftFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_FRONT_TALON);
+    rightRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_REAR_TALON);
+    rightMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_MIDDLE_TALON);
+    rightFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_FRONT_TALON);
+    drivetrain = new DifferentialDrive(new SpeedControllerGroup(leftRearTalon, leftMiddleTalon, leftFrontTalon),
+        new SpeedControllerGroup(rightRearTalon, rightMiddleTalon, rightFrontTalon));
+    // TODO: set correct port for pigeon gyro.
+    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
+    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
+    //TODO: set correct talons for encoders.
+    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
+    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);
+  }
+
+  // Drive functions
+  public void arcadeDrive(double x, double y) {
+    drivetrain.arcadeDrive(x, y);
+  }
+
+  public void curvatureDrive(double x, double y, boolean quickTurn) {
+    drivetrain.curvatureDrive(x, y, quickTurn);
+  }
+
+  public void tankDrive(double leftSpeed, double rightSpeed) {
+    drivetrain.tankDrive(leftSpeed, rightSpeed);
+  }
+
+  public void voltageTankDrive(double left, double right) {
+    tankDrive(left / RobotController.getBatteryVoltage(), right / RobotController.getBatteryVoltage());
+  }
+
+  // Gyro functions
+  double getAngle() {
+    return gyro.getAngle();
+  }
+
+  public double getRadianAngle() {
+    return Math.toRadians(getAngle());
```

Comment:
replace `Math.toRadians(getAngle())` with `Math.toRadians(gyro.getAngle());` . Since we don't want to use Math.IEEE... on this function